packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts

```diff
@@ -0,0 +1,61 @@
+import { useModalAttributes } from '@fluentui/react-tabster';
+import { applyTriggerPropsToChildren, getTriggerChild, useEventCallback } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { useDialogContext_unstable } from '../../contexts/dialogContext';
+import { isTargetDisabled } from '../../utils/isTargetDisabled';
+import {
+  DialogTriggerChildProps,
+  DialogTriggerProps,
+  DialogTriggerState,
+  DialogTriggerAction,
+} from './DialogTrigger.types';
+
+/**
+ * Create the state required to render DialogTrigger.
+ * Clones the only child component and adds necessary event handling behaviours to open a popup Dialog
+ *
+ * @param props - props from this instance of DialogTrigger
+ */
+export const useDialogTrigger_unstable = (props: DialogTriggerProps): DialogTriggerState => {
+  const { children, action = 'toggle' } = props;
+
+  const child = React.isValidElement(children) ? getTriggerChild(children) : undefined;
+
+  const requestOpenChange = useDialogContext_unstable(ctx => ctx.requestOpenChange);
+
+  const { triggerAttributes } = useModalAttributes();
+
+  const handleClick = useEventCallback((event: React.MouseEvent<HTMLElement>) => {
```

Comment:
should `keydown` be handled aswell ? With the trigger pattern there's no guarantee that the child element is a `button` or `a` which support native keyboard mapping to the click handler